uswds/uswds-site
# USWDS-Site - Prettier should run on .md files.

# Summary

We were ignoring running prettier on markdown files. This can cause inconsistencies and errors when writing MD/Frontmatter.

Example: https://github.com/uswds/uswds-site/commit/1c6453d5d19af098d10294b88a44b62517bddf40

## Related issue

Note comment:

https://github.com/uswds/uswds-site/issues/2868#issuecomment-2435998529

These are links to these files, there weren't any problems running prettier

[pages/design-principles/overview.md](https://github.com/uswds/uswds-site/blob/main/pages/design-principles/overview.md) (before)
[pages/design-principles/overview.md](https://github.com/uswds/uswds-site/blob/cb-enable-prettier-on-markdown/pages/design-principles/overview.md) (after)

https://github.com/uswds/uswds-site/blob/main/pages/documentation/maturity-model.md (before)
https://github.com/uswds/uswds-site/blob/cb-enable-prettier-on-markdown/pages/documentation/maturity-model.md (after)

Closes #2868

## Problem statement

By not running prettier against .md files (in .prettierignore), we are not ensuring these files are subject to coding standards and proper formatting.

## Solution

1. Remove .md from .prettierignore.
2. Create two new NPM script to check & fix.
4. Create CI test for linting and formatting.
Fix issues.

## Major changes

Added new script to config.yml:
 - run:
          name: Check formatting
          command: npm run prettier:check

Prettier check in package.json now checks .md files

lint in package.json runs "npx gulp lintJS lintSass && npm run prettier",

run prettier runs both prettier:md && prettier:scss

Do not run prettier on the vendor directory since we don't manage those md files.

Pipeline before running prettier:

![image](https://github.com/user-attachments/assets/b1094a51-72dc-4ae2-b284-ef7e3d12d983)

Pipeline after running prettier:

![image](https://github.com/user-attachments/assets/186b9f0b-df82-4b3a-9651-733589449f8d)

## Testing and review

1. Pull the branch. 
2. Run prettier.
3. Notice the changes to the markdown files.
4. Commit the changed files and push.
5. Make sure the pipeline doesn't fail because of prettier.
